airflow/sensors/http_sensor.py

```diff
@@ -81,8 +89,13 @@ def poke(self, context):
                                      headers=self.headers,
                                      extra_options=self.extra_options)
             if self.response_check:
+
+                response_check_kwargs = {}
+                if self.provide_context:
+                    response_check_kwargs["context"] = context
+
                 # run content check on response
-                return self.response_check(response)
+                return self.response_check(response, **response_check_kwargs)
```

Comment:
```suggestion
                return self.response_check(response, context)
```
Would do the trick as well, right? This will break backward compatibility because if the context is undefined, this won't work.

I think we can make this nicer:
```python
if self.response_check:
  if self.provide_context:
    return self.response_check(response, context)
  else:
    return self.response_check(response)
```
WDYT?

Reply:
Yes we can do this